build-configuration/android-application.gradle

```diff
@@ -0,0 +1,64 @@
+/**
+ * Base build configuration for all android example applications (modules applying the android application plugin).
+ *
+ * Don't add any configuration here that is not common to *every* android example application.
+ */
+apply plugin: "com.android.application"
+apply plugin: "kotlin-android"
+
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion rootProject.ext.compileSdkVersion
+        versionCode 11
```

Comment:
we need to manually bump up the version code of the identity example app when a new build is pushed to play console, with this overriding all example apps, all apps would need to be on the same version code and bumped together, is that expected?

Reply:
I'm fine with bumping them all together, or I can keep the versionCode in the identity-example module, and that will override this generic one for all, or I can add them all back to their respective build.gradle files.

Let me know your preference and I'll update accordingly.

Reply:
Should example apps match our SDK versioning?

here we could do:

```
versionName VERSION_NAME
versionCode getVersionCode(VERSION_NAME)
```

Where getVersionCode coverts our version name to a comparable integer.

```java
// versions.gradle
def getVersionCode(versionName) {
    // Split the version name by dots into an array of integers
    def versionNumbers = versionName.split('\\.').collect { it.toInteger() }

    // Calculate the version code as a sum of the version numbers
    return versionNumbers[0] * 10000 + versionNumbers[1] * 100 + versionNumbers[2]
}
```

Reply:
That's another option that works for me!

Reply:
that works for me! yeah would be great to avoid updating the actual version number every time.

Now that all the apps are on the same version number, we could probably add some automatic job to push them to play console periodically(not required in this PR)

Reply:
hmm sometimes we'll need to publish arbitrary number of builds to play console(e.g during a bugbash we might want to update the app a couple of times per week to the testers), this would prevent us from doing that, could we instead let each example app define their own version code?

Or I can override the value for identity only